### PR TITLE
Improve logging performance by checking Trace/Debug log level is enabled

### PR DESCRIPTION
**Pull request by nzdev**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes 

### Description

Improve logging performance by checking Trace/Debug log level is enabled before trying to call the logger. This saves on method calls and allocations.

How to test?
Existing tests should pass. Performance  should be improved (lower allocations, faster response times, lower cpu usage)

### CODE REVIEW

**Comment by emmagarland** [comment]
Hi again @nzdev!

Just going through some of the longer running PRs again, and noticed this one. Sorry for the delay, there are now a few conflicts I'm afraid!

Does this still stand as something that you'd like to work on? If so, I can help to progress it further.

Thanks,

Emma 🙂

File: src/Umbraco.Core/Composing/ComponentCollection.cs [hunk 1]
```diff
@@ -23,13 +23,13 @@ public ComponentCollection(Func<IEnumerable<IComponent>> items, IProfilingLogger
 
     public void Initialize()
     {
-        using (_profilingLogger.DebugDuration<ComponentCollection>(
+        using (_profilingLogger.IsEnabled(Logging.LogLevel.Debug) ? null : _profilingLogger.DebugDuration<ComponentCollection>(
```
Comment (nul800sebastiaan) [inline src/Umbraco.Core/Composing/ComponentCollection.cs:26]: Am I reading this wrong: if `LogLevel.Debug` is enabled, don't log (`? null`) else, do log. Did you mean to negate the check? `!_profilingLogger.IsEnabled(Logging.LogLevel.Debug)`

**Comment by nul800sebastiaan** [comment]
Just to note, one of end-to-end tests was failing for unrelated reasons (`##[error]The job running on agent Azure Pipelines 2 ran longer than the maximum time of 60 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134`). The build is fine otherwise. 

I think there's a few places @nzdev where you're not doing the opposite of what is expected, `null` instead of actually logging things, if you could fix those up then we can continue the review! 👍

File: src/Umbraco.Core/Composing/ComponentCollection.cs (hunk 1 above)
Comment (nzdev) [reply src/Umbraco.Core/Composing/ComponentCollection.cs]: Makes sense. I assume I copied and pasted this. Fixed up.

Comment (nul800sebastiaan) [reply src/Umbraco.Core/Composing/ComponentCollection.cs]: @nzdev Thanks! Same problem in 18 other files though! 😉 Anywhere you used a ternary, it's the wrong way around. 😅

Comment (nul800sebastiaan) [reply src/Umbraco.Core/Composing/ComponentCollection.cs]: Ah wait, I see there's 2 commits, I only saw the last one, I think that covers all of them 👍

**Review by nul800sebastiaan — CHANGES_REQUESTED** [review]
Found a few more and then I think this is good to go!

**Review by nul800sebastiaan — APPROVED** [review]

**Comment by nul800sebastiaan** [comment]
Thanks again @nzdev merged and ready to go for 12.1, great work! 👍

**Pull request merged**